### ISSUE TITLE
BUG: Multi-gpu (DDP) with Pytorch Lightning creates a duplicate runs for each gpu 

### ISSUE THREAD

**Issue by m-bain** [neptune-ai/neptune-client#757]
### Describe the bug
When using neptune with pytorch lightning in multi-gpu ddp duplicate runs for each gpu are created.
### Reproduction
Minimal example:
```
import pytorch_lightning as pl
from pytorch_lightning.loggers.neptune import NeptuneLogger
from torchvision.datasets import MNIST
from torchvision import transforms
from src.models.mnist_model import MNISTLitModel
from torch.utils.data import DataLoader
import torch
import torch.nn.functional as F

PATH_DATASETS = "./"
AVAIL_GPUS = 4
BATCH_SIZE = 256


class MNISTModel(pl.LightningModule):
    def __init__(self):
        super().__init__()
        self.l1 = torch.nn.Linear(28 * 28, 10)

    def forward(self, x):
        return torch.relu(self.l1(x.view(x.size(0), -1)))

    def training_step(self, batch, batch_nb):
        x, y = batch
        loss = F.cross_entropy(self(x), y)
        return loss

    def configure_optimizers(self):
        return torch.optim.Adam(self.parameters(), lr=0.02)


# Create NeptuneLogger
neptune_logger = NeptuneLogger(
    project="name/PROJECT")

train_ds = MNIST(PATH_DATASETS, train=True, download=True, transform=transforms.ToTensor())
train_loader = DataLoader(train_ds, batch_size=BATCH_SIZE)

# Pass NeptuneLogger to the Trainer
# Initialize a trainer
trainer = pl.Trainer(
    gpus=AVAIL_GPUS,
    max_epochs=3,
    progress_bar_refresh_rate=20,
    logger=neptune_logger,
    strategy='ddp'
)

# Train the model ⚡
trainer.fit(MNISTLitModel(), train_loader)
```
### Expected behavior
Only one run created...
### Traceback
OUTPUT (run per gpu as shown below):
```
https://app.neptune.ai/name/PROJECT/e/LON-46
initializing distributed: GLOBAL_RANK: 1, MEMBER: 2/4
initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/4
https://app.neptune.ai/name/PROJECT/e/LON-47
initializing distributed: GLOBAL_RANK: 2, MEMBER: 3/4
https://app.neptune.ai/name/PROJECT/e/LON-48
initializing distributed: GLOBAL_RANK: 3, MEMBER: 4/4
```
### Environment
**The output of `pip list`:**
Package                           Version
--------------------------------- ------------
absl-py                           1.0.0
aiohttp                           3.8.0
aiosignal                         1.2.0
alembic                           1.7.5
antlr4-python3-runtime            4.8
anyio                             3.3.4
argon2-cffi                       21.1.0
async-timeout                     4.0.1
attrs                             21.2.0
autopage                          0.4.0
Babel                             2.9.1
backcall                          0.2.0
backports.entry-points-selectable 1.1.1
black                             21.10b0
bleach                            4.1.0
boto3                             1.20.7
botocore                          1.23.7
bravado                           11.0.3
bravado-core                      5.17.0
cachetools                        4.2.4
certifi                           2021.10.8
cffi                              1.15.0
cfgv                              3.3.1
charset-normalizer                2.0.7
click                             8.0.3
cliff                             3.9.0
cmaes                             0.8.2
cmd2                              2.3.0
colorama                          0.4.4
colorlog                          6.6.0
commonmark                        0.9.1
configparser                      5.1.0
cycler                            0.11.0
debugpy                           1.5.1
decorator                         5.1.0
decord                            0.6.0
defusedxml                        0.7.1
distlib                           0.3.3
docker-pycreds                    0.4.0
entrypoints                       0.3
filelock                          3.3.2
flake8                            4.0.1
frozenlist                        1.2.0
fsspec                            2021.11.0
future                            0.18.2
gitdb                             4.0.9
GitPython                         3.1.24
google-auth                       2.3.3
google-auth-oauthlib              0.4.6
greenlet                          1.1.2
grpcio                            1.41.1
huggingface-hub                   0.1.2
hydra-colorlog                    1.1.0
hydra-core                        1.1.0
hydra-optuna-sweeper              1.1.0
identify                          2.3.5
idna                              3.3
importlib-metadata                4.8.2
importlib-resources               5.4.0
iniconfig                         1.1.1
ipykernel                         6.5.0
ipython                           7.29.0
ipython-genutils                  0.2.0
isort                             5.10.1
jedi                              0.18.0
Jinja2                            3.0.3
jmespath                          0.10.0
joblib                            1.1.0
json5                             0.9.6
jsonpointer                       2.2
jsonref                           0.2
jsonschema                        3.2.0
jupyter-client                    7.0.6
jupyter-core                      4.9.1
jupyter-server                    1.11.2
jupyterlab                        3.2.3
jupyterlab-pygments               0.1.2
jupyterlab-server                 2.8.2
kiwisolver                        1.3.2
Mako                              1.1.5
Markdown                          3.3.4
MarkupSafe                        2.0.1
matplotlib                        3.4.3
matplotlib-inline                 0.1.3
mccabe                            0.6.1
mistune                           0.8.4
monotonic                         1.6
msgpack                           1.0.2
multidict                         5.2.0
mypy-extensions                   0.4.3
nbclassic                         0.3.4
nbclient                          0.5.8
nbconvert                         6.3.0
nbformat                          5.1.3
neptune-client                    0.13.1
nest-asyncio                      1.5.1
nodeenv                           1.6.0
notebook                          6.4.5
numpy                             1.19.5
oauthlib                          3.1.1
omegaconf                         2.1.1
optuna                            2.4.0
packaging                         21.2
pandas                            1.3.4
pandocfilters                     1.5.0
parso                             0.8.2
pathspec                          0.9.0
pathtools                         0.1.2
pbr                               5.7.0
pexpect                           4.8.0
pickleshare                       0.7.5
Pillow                            8.4.0
pip                               21.2.4
platformdirs                      2.4.0
pluggy                            1.0.0
pre-commit                        2.15.0
prettytable                       2.4.0
prometheus-client                 0.12.0
promise                           2.3
prompt-toolkit                    3.0.22
protobuf                          3.19.1
psutil                            5.8.0
ptyprocess                        0.7.0
pudb                              2021.2.2
py                                1.11.0
pyasn1                            0.4.8
pyasn1-modules                    0.2.8
pycodestyle                       2.8.0
pycparser                         2.21
pyDeprecate                       0.3.1
pyflakes                          2.4.0
Pygments                          2.10.0
PyJWT                             2.3.0
pyparsing                         2.4.7
pyperclip                         1.8.2
pyrsistent                        0.18.0
pytest                            6.2.5
python-dateutil                   2.8.2
python-dotenv                     0.19.2
pytorch-lightning                 1.5.1
pytz                              2021.3
PyYAML                            6.0
pyzmq                             22.3.0
regex                             2021.11.10
requests                          2.26.0
requests-oauthlib                 1.3.0
rfc3987                           1.3.8
rich                              10.13.0
rsa                               4.7.2
s3transfer                        0.5.0
sacremoses                        0.0.46
scikit-learn                      1.0.1
scipy                             1.7.2
seaborn                           0.11.2
Send2Trash                        1.8.0
sentencepiece                     0.1.96
sentry-sdk                        1.4.3
setuptools                        58.0.4
sh                                1.14.2
shortuuid                         1.0.8
simplejson                        3.17.6
six                               1.16.0
smmap                             5.0.0
sniffio                           1.2.0
SQLAlchemy                        1.4.27
stevedore                         3.5.0
strict-rfc3339                    0.7
subprocess32                      3.5.4
swagger-spec-validator            2.7.4
tensorboard                       2.7.0
tensorboard-data-server           0.6.1
tensorboard-plugin-wit            1.8.0
termcolor                         1.1.0
terminado                         0.12.1
testpath                          0.5.0
threadpoolctl                     3.0.0
tokenizers                        0.10.3
toml                              0.10.2
tomli                             1.2.2
torch                             1.10.0+cu113
torchaudio                        0.10.0+cu113
torchmetrics                      0.6.0
torchvision                       0.11.1+cu113
tornado                           6.1
tqdm                              4.62.3
traitlets                         5.1.1
transformers                      4.12.3
typing-extensions                 3.10.0.2
urllib3                           1.26.7
urwid                             2.1.2
urwid-readline                    0.13
virtualenv                        20.10.0
wandb                             0.12.6
wcwidth                           0.2.5
webcolors                         1.11.1
webencodings                      0.5.1
websocket-client                  1.2.1
Werkzeug                          2.0.2
wheel                             0.37.0
yarl                              1.7.2
yaspin                            2.1.0
zipp                              3.6.0

**The operating system you're using:**
```NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

**The output of `python --version`:**
Python 3.8.12

**Comment by m-bain** [comment]
to be solved in pytorch-lightning
